modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2142,23 +2143,35 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
     # END Map across rows/columns
 
     # __getitem__ methods
+    __getitem_bool = Binary.register(
+        lambda df, r: df[r], join_type="left", labels="drop"
+    )
+
+    def __validate_bool_indexer(self, indexer):
+        if len(indexer) != len(self.index):
+            raise ValueError(
+                f"Item wrong length {len(indexer)} instead of {len(self.index)}."
+            )
+        if isinstance(indexer, pandas.Series) and not indexer.equals(self.index):
+            warnings.warn(
+                "Boolean Series key will be reindexed to match DataFrame index.",
+                PendingDeprecationWarning,
```

Comment:
What is the need in warning type?  (`PendingDeprecationWarning`)

Reply:
I just moved the code around, I think this is what pandas produces?..
https://github.com/pandas-dev/pandas/blob/a046792de912146582828de1baf3008750d93bd0/pandas/core/frame.py#L3771-L3776